### PR TITLE
Bugfix: Class-agnostic mAP

### PR DESCRIPTION
**Pull request by AHuzail**
# Description

This pull request fixes the issue with the class_agnostic argument in the `MeanAveragePrecision` class, which was previously not functioning correctly. `The class_agnostic` flag now treats all predictions and ground truth as belonging to the same class when set to `True`. Added a `_match_detection_batch_class_agnostic` functionality.

[link for collab](https://colab.research.google.com/drive/10EaQ4lJNXzcmub7doO2vGn2t1LgDdkcC?usp=sharing)
## Type of change

Please delete options that are not relevant.

-  [X] Bug fix (non-breaking change which fixes an issue)

## How has this change been tested, please provide a testcase or example of how you tested the change?

This fix has been thoroughly tested using newly added unit tests, covering the following scenarios:

1. **Class-Specific mAP Calculation**:
   - Verified that mAP is computed correctly when `class_agnostic` is set to `False` (default).
   
2. **Class-Agnostic mAP Calculation**:
   - Confirmed that mAP is calculated without class distinction when `class_agnostic=True`.
  
3. **Edge Cases**:
   - Included test cases for no overlaps, perfect match scenarios, and partial overlaps.
## Any specific deployment considerations

No

## Docs

-   [x] Docs updated? What were the changes: No

### CODE REVIEW

**Comment by LinasKo** [comment]
Thanks @AHuzail! I'll review it this week.

File: supervision/metrics/mean_average_precision.py [hunk 1]
```diff
@@ -75,6 +75,13 @@ def update(
                 f"The number of predictions ({len(predictions)}) and"
                 f" targets ({len(targets)}) during the update must be the same."
             )
+        # class-agnostic
```
[the remark this answers is not in the document]
Comment (AHuzail) [reply supervision/metrics/mean_average_precision.py]: Removed comment

File: supervision/metrics/mean_average_precision.py
[the remark this answers is not in the document]
Comment (AHuzail) [reply supervision/metrics/mean_average_precision.py:1]: done

**Comment by LinasKo** [comment]
Nice one, that was quick! I'll have a look tomorrow

**Review by LinasKo — APPROVED** [review]
Fantastic work! Thank you very much, and welcome to the community 😉
